src/internet_identity/internet_identity.did

```diff
@@ -145,6 +157,18 @@ type IdentityAnchorInfo = record {
     device_registration: opt DeviceRegistrationInfo;
 };
 
+type DeployArchiveResult = variant {
+  // The archive was deployed successfully and the supplied wasm module has been installed. The principal of the archive
+  // canister is returned.
+  success: principal;
```

Comment:
This is because you cannot create the archive but not deploy it. So it is either:
* create and deploy
* just deploy

Thus, having a single success result at the end of all actions made sense to me...